fineract-provider/src/main/java/org/apache/fineract/portfolio/address/api/AddressConfigurationApiResources.java

```diff
@@ -0,0 +1,112 @@
+import org.apache.fineract.infrastructure.core.api.ApiRequestParameterHelper;
+import org.apache.fineract.infrastructure.core.serialization.ApiRequestJsonSerializationSettings;
+import org.apache.fineract.infrastructure.core.serialization.DefaultToApiJsonSerializer;
+import org.apache.fineract.infrastructure.security.service.PlatformSecurityContext;
+import org.apache.fineract.portfolio.address.data.AddressData;
+import org.apache.fineract.portfolio.address.data.FieldConfigurationData;
+import org.apache.fineract.portfolio.address.service.AddressReadPlatformServiceImpl;
+import org.apache.fineract.portfolio.address.service.FieldConfigurationReadPlatformService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+@Path("/clients/addresses")
+@Component
+@Scope("singleton")
+public class AddressConfigurationApiResources {
+	private final Set<String> RESPONSE_DATA_PARAMETERS = new HashSet<>(Arrays.asList("clientAddressId", "client_id",
+			"address_id", "address_type_id", "is_active", "fieldConfigurationId", "entity", "table", "field",
+			"is_enabled", "is_mandatory", "validation_regex"));
+	private final String resourceNameForPermissions = "Address";
+	private final PlatformSecurityContext context;
+	private final AddressReadPlatformServiceImpl readPlatformService;
+	private final DefaultToApiJsonSerializer<AddressData> toApiJsonSerializer;
+	private final FieldConfigurationReadPlatformService readPlatformServicefld;
+	private final DefaultToApiJsonSerializer<FieldConfigurationData> toApiJsonSerializerfld;
+	private final ApiRequestParameterHelper apiRequestParameterHelper;
+	private final PortfolioCommandSourceWritePlatformService commandsSourceWritePlatformService;
+
+	@Autowired
+	public AddressConfigurationApiResources(final PlatformSecurityContext context,
+			final AddressReadPlatformServiceImpl readPlatformService,
+			final DefaultToApiJsonSerializer<AddressData> toApiJsonSerializer,
+			final FieldConfigurationReadPlatformService readPlatformServicefld,
+			final DefaultToApiJsonSerializer<FieldConfigurationData> toApiJsonSerializerfld,
+			final ApiRequestParameterHelper apiRequestParameterHelper,
+			final PortfolioCommandSourceWritePlatformService commandsSourceWritePlatformService) {
+		this.context = context;
+		this.readPlatformService = readPlatformService;
+		this.toApiJsonSerializer = toApiJsonSerializer;
+		this.readPlatformServicefld = readPlatformServicefld;
+		this.toApiJsonSerializerfld = toApiJsonSerializerfld;
+		this.apiRequestParameterHelper = apiRequestParameterHelper;
+		this.commandsSourceWritePlatformService = commandsSourceWritePlatformService;
+	}
+
+	@GET
+	@Path("/fieldconfiguration/{entity}")
+	@Consumes({ MediaType.APPLICATION_JSON })
+	@Produces({ MediaType.APPLICATION_JSON })
+	public String getAddresses(@PathParam("entity") final String entityname, @Context final UriInfo uriInfo) {
```

Comment:
Already EntityFieldConfigurationApiResources.getAddresses(...) API is returning same Data. Why this API is required in this class?